src/lib/evaluate.ts

```diff
@@ -178,8 +228,129 @@ export async function evaluatePromptOnTestSuite(params: {
     }
   }
   return {
-    success: passedTests === testCases.length,
+    success: passedTests == testCases.length,
     passedTests,
     totalTests: testCases.length,
+    totalUsage,
+  };
+}
+
+/**
+ * Compresses a given text using a model and a prompt
+ * @param params Parameters for compressing text
+ * @returns The compressed text and the token usage for the compression
+ */
+async function getCompressedTask(params: {
+  task: string;
+  compressingPrompt: string;
+  model: string;
+}): Promise<{ compressedTask: string; usage: OpenAI.CompletionUsage }> {
+  const { task, compressingPrompt, model } = params;
+  const openai = createOpenAIClient();
+  await new Promise(r => setTimeout(r, 500)); // Rate limit friendly
```

Comment:
Do not do delays. If we face rate limiting, we will add retries with exponential debouncing.

Reply:
https://www.npmjs.com/package/p-retry is good for that, but only use it if we really hit rate limiting from openrouter (most probably we will not).